packages/carbon-graphs/src/js/controls/Graph/Graph.js

```diff
@@ -426,64 +426,63 @@ class Graph extends Construct {
+              data.values.forEach((v) => this.config.axis.x.ticks.values.push(v.x));
+            }
+            this.content[position].reflow(this, data);
+            setAxisPadding(this.config.axisPadding, this.content[position]);
+            getAxesDataRange(
+              this.content[position],
+              this.content[position].config.yAxis,
+              this.config,
+              this.content,
+            );
+            if (
+              this.config.allowCalibration
+                && isRangeModified(
+                  this.config,
+                  this.content[position].config.yAxis,
+                )
+            ) {
+              updateAxesDomain(this.config, this.content[position]);
+            }
+            if (
+              this.config.showNoDataText
+                && this.content.every((content) => utils.isEmpty(content.config.values))
+            ) {
+              drawNoDataView(this.config, this.svg);
+              redrawDatelineContent(this.scale, this.config, this.svg);
+            } else if (utils.notEmpty(this.content[position].config.values)) {
+            // Removes existing No Data View, when legend hold values
+              removeNoDataView(this.svg);
+            }
+          }
+          position = -1;
         }
-      }
+      });
     }
 
-    updateXAxisDomain(this.config);
-    scaleGraph(this.scale, this.config);
-    translateAxes(this.axis, this.scale, this.config, this.svg);
-
-    if (graphData && graphData.eventline) {
+    if(graphData && graphData.eventline) {
       this.config.eventline=graphData.eventline;
       redrawEventlineContent(this.scale, this.config, this.svg);
     }
 
-    if (
-      graphData &&
-      graphData.values &&
-      this.contentKeys.includes(graphData.key)
-    ) {
-      this.content[position].reflow(this, graphData);
-      setAxisPadding(this.config.axisPadding, this.content[position]);
-      getAxesDataRange(
-        this.content[position],
-        this.content[position].config.yAxis,
-        this.config,
-        this.content,
-      );
-      if (
-        this.config.allowCalibration
-                && isRangeModified(
-                  this.config,
-                  this.content[position].config.yAxis,
-                )
-      ) {
-        updateAxesDomain(this.config, this.content[position]);
-      }
-      if (
-        this.config.showNoDataText
-                && this.content.every((content) => utils.isEmpty(content.config.values))
-      ) {
-        drawNoDataView(this.config, this.svg);
-        redrawDatelineContent(this.scale, this.config, this.svg);
-      } else if (utils.notEmpty(this.content[position].config.values)) {
-        // Removes existing No Data View, when legend hold values
-        removeNoDataView(this.svg);
-      }
-    }
+    updateXAxisDomain(this.config);
+    scaleGraph(this.scale, this.config);
+    translateAxes(this.axis, this.scale, this.config, this.svg);
+
```

Comment:
Does changing the order of calling these methods makes any difference?

Reply:
The conditions required for updating the ticks and the data were the same. So, instead of having redundant checks for both the tasks, I thought of clubbing them under 1 `if` block.